ggircs-app/app/components/XmlDiff/RenderDiff.tsx

```diff
@@ -1,35 +1,64 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useMemo } from "react";
+import { graphql, createFragmentContainer } from "react-relay";
+import { RenderDiff_query } from "RenderDiff_query.graphql";
 import { diffLines, formatLines } from "unidiff";
 import { parseDiff, Diff, Hunk, tokenize } from "react-diff-view";
 import "react-diff-view/style/index.css";
+import format from "xml-formatter";
+import { NextRouter } from "next/router";
 import LoadingSpinner from "components/LoadingSpinner";
 
 const EMPTY_HUNKS = [];
 
 interface Props {
-  oldText: string;
-  newText: string;
-  collapse: boolean;
+  query: RenderDiff_query;
+  router: NextRouter;
 }
 
 export const RenderDiff: React.FunctionComponent<Props> = ({
-  oldText,
-  newText,
-  collapse,
+  query,
+  router,
 }) => {
-  const [loaded, setLoaded] = useState(false);
+  const shouldRenderDiff =
+    query.firstSideReport &&
+    query.secondSideReport &&
+    router.query.FirstSideRelayId &&
+    router.query.SecondSideRelayId;
+  if (!shouldRenderDiff) return null;
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [diffText, setDiffText] = useState(null);
+
+  const oldText = format(
+    query.firstSideReport.latestSwrsReport.ecccXmlFileByEcccXmlFileId.xmlFile
+  );
+  const newText = format(
+    query.secondSideReport.latestSwrsReport.ecccXmlFileByEcccXmlFileId.xmlFile
+  );
 
-  // UseEffect & setState to determine if this slow component is loading. Renders the loading spinner while loading.
   useEffect(() => {
-    setLoaded(true);
-  }, []);
+    setIsLoading(false);
+  });
+
+  useMemo(() => {
+    setIsLoading(true);
+    if (router.query.reversed)
+      setDiffText(
+        formatLines(diffLines(newText, oldText), {
+          // context is the max number of lines to show around a change. When collapsed, we set it to 1. Otherwise 10000 to ensure we get the whole document.
+          context: router.query.collapsed ? 1 : 10000,
+        })
+      );
+    else
+      setDiffText(
+        formatLines(diffLines(oldText, newText), {
+          context: router.query.collapsed ? 1 : 10000,
```

Comment:
Maybe we could make this values configurable, for example in the props? Or declare it once on top of the file at least